src/Proto.Cluster/ClusterExtension.cs

```diff
@@ -27,26 +27,30 @@ public static Cluster Cluster(this ActorSystem system)
         public static Cluster Cluster(this IContext context)
             => context.System.Extensions.Get<Cluster>() ?? throw new NotSupportedException("Cluster has not been configured");
 
-        public static Task<T> ClusterRequestAsync<T>(this IContext context, string identity, string kind, object message, CancellationToken ct)
-        {
-            var cluster = context.System.Extensions.Get<Cluster>();
+        public static Task<T> ClusterRequestAsync<T>(this IContext context, string identity, string kind, object message, CancellationToken ct) =>
             //call cluster RequestAsync using actor context
-            return cluster.RequestAsync<T>(identity, kind, message, context, ct);
-        }
-
-        public static Props WithClusterInit(this Props props, Cluster cluster, ClusterIdentity clusterIdentity, ActivatedClusterKind activatedClusterKind)
+            context.System.Cluster()!.RequestAsync<T>(identity, kind, message, context, ct);
+
+        public static Props WithClusterInit(
+            this Props props,
+            Cluster cluster,
+            ClusterIdentity clusterIdentity,
+            ActivatedClusterKind activatedClusterKind
+        )
         {
-            return props.WithReceiverMiddleware(
-                baseReceive =>
-                    (ctx, env) => {
-                        return env.Message switch
-                        {
-                            Started => HandleStart(baseReceive, ctx, env),
-                            Stopped => HandleStopped(baseReceive, ctx, env),
-                            _       => baseReceive(ctx, env)
-                        };
-                    }
-            );
+            return props
+                .WithOnInit(context => context.Set(clusterIdentity))
```

Comment:
I really like how this turned out, super smooth and aligns well with rest of API